src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.Editor.cs

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.RemoveUnnecessaryImports;
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.MoveType
+{
+    internal abstract partial class AbstractMoveTypeService<TService, TTypeDeclarationSyntax, TNamespaceDeclarationSyntax, TMemberDeclarationSyntax, TCompilationUnitSyntax>
+    {
+        private class Editor
+        {
+            private readonly CancellationToken _cancellationToken;
+            private readonly MoveTypeOptionsResult _moveTypeOptions;
+            private readonly State _state;
+            private readonly TService _service;
+
+            private readonly bool _fromDialog;
+            private readonly bool _makeOuterTypesPartial;
+            private readonly bool _makeTypePartial;
+            private readonly bool _renameFile;
+            private readonly bool _renameType;
+
+            public Editor(
+                TService service,
+                State state,
+                bool renameFile,
+                bool renameType,
+                bool makeTypePartial,
+                bool makeOuterTypePartial,
+                MoveTypeOptionsResult moveTypeOptions,
+                bool fromDialog,
+                CancellationToken cancellationToken)
+            {
+                _renameFile = renameFile;
+                _makeTypePartial = makeTypePartial;
+                _makeOuterTypesPartial = makeOuterTypePartial || makeTypePartial;
+                _renameType = renameType;
+                _state = state;
+                _service = service;
+                this._moveTypeOptions = moveTypeOptions;
+                this._fromDialog = fromDialog;
+                this._cancellationToken = cancellationToken;
+            }
+
+            private SemanticDocument SemanticDocument => _state.SemanticDocument;
+
+            internal async Task<IEnumerable<CodeActionOperation>> GetOperationsAsync()
+            {
+                var solution = SemanticDocument.Document.Project.Solution;
+
+                if (_renameFile)
+                {
+                    return RenameFileToMatchTypeName(solution);
+                }
+                else if (_renameType)
+                {
+                    return await RenameTypeToMatchFileAsync(solution).ConfigureAwait(false);
+                }
+
+                var documentName = _fromDialog
+                    ? _moveTypeOptions.NewFileName
```

Comment:
I feel like at thsi point hte moveTypeOptions should have been already incorporated.  i.e. you shouldn't care right here if it was _fromDialog or not.  The person who made the dialog should have extracted the values and called into this with the right data.